roadmap/implementers-guide/src/node/dispute/votesdb.md

```diff
@@ -0,0 +1,139 @@
+# Votes DB
+
+Responsible module for tracking all kind of dispute votes by all validators
+for a fixed number of sessions.
+
+
+## Remarks
+
+Votes must be persisted. Disputes might happen long after
+there was a dispute, as such just keeping things in memory
+is thus insufficient, as every blip (OOM, bug, maintanance)
+could cause the node to forget the state.
+
+`UnsignedTransaction`s are OK to be used, since the inner
+element `CommittedCanddidateReceipt` is verifiable.
+
+## ToDo
+
+* Missing Definition of Governance Mode.
+* Is a `PoV` enough or should the whole `CommittedCandidateReceipt` be included?
+* Currently [`fn mark_bad()`](https://github.com/paritytech/substrate/pull/6301/files#diff-8faeb5c685a8fdff428c5ec6d9102fd59e127ff69762d43045cd38e586db5559R60-R64) does not persist data.
+
+## IO
+
+Inputs:
+
+* `VotesDbMessage::`
+
+Outputs:
+
+* `DisputeConclusionMessage::Detection`
+* `DisputeConclusionMessage::Resolution`
+* `ChainApiMessage::Blacklist`
```

Comment:
Why is this the module that handles `Blacklist`?

Reply:
It seemed reasonable, that once the dispute is resolved, to send out the blacklist from the same module. But the `ChainApiMessage` is wrong anyways, since the its a client API to revert and blacklist blocks.